servicetalk-concurrent-api/src/main/java/io/servicetalk/concurrent/api/BufferStrategies.java

```diff
@@ -279,9 +286,14 @@ void beforeNewAccumulatorEmitted(final Accumulator<T, B> newAccumulator) {
 
         @Override
         public void accumulate(@Nullable final T item) {
-            size++;
+            if (sizeThreshold > 0 || isEmpty()) {
+                // Increment `size` only when the `sizeThreshold` is used or for the first accumulation
```

Comment:
Can we simplify this logic and just unconditionally increment? The conditional logic is more involved to understand how it relates to the other state and the `delegate`.

Reply:
Ok, I will keep the `size = addWithOverflowProtection(size, 1);` then

Reply:
Discussed offline, this does not require any protection due to how we manage the demand for items publisher. Changed to `++size`.